src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java

```diff
@@ -86,16 +86,68 @@
  *    &lt;property name="format" value="^[a-z](_?[a-zA-Z0-9]+)*$"/&gt;
  * &lt;/module&gt;
  * </pre>
- *
+ * <p>Code Example:</p>
+ * <pre>
+ * class MyClass {
+ *   public void myMethod() {} // OK
+ *   public void MyMethod() {} // violation, name "MyMethod"
+ *                             // should match the pattern "^[a-z](_?[a-zA-Z0-9]+)*$"
+ * }
+ * </pre>
  * <p>
- * An example of how to configure the check to allow method names
- * to be equal to the residing class name is:
+ * An example of how to configure the check to allow method names to be equal to the
+ * residing class name is:
  * </p>
  * <pre>
  * &lt;module name="MethodName"&gt;
+ *    &lt;property name="format" value="^[a-zA-Z](_?[a-zA-Z0-9]+)*$"/&gt;
  *    &lt;property name="allowClassName" value="true"/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <p>Code Example:</p>
+ * <pre>
+ * class MyClass {
+ *   public MyClass() {} // OK
+ *   public void MyClass() {} // OK, method Name 'MyClass' is allowed to be
+ *                            // equal to the enclosing class name
+ * }
+ * </pre>
+ * <p>
+ * A counter example of the allowClassName which its value is "false":
+ * </p>
+ * <pre>
+ * &lt;module name="MethodName"&gt;
+ *    &lt;property name="format" value="^[a-zA-Z](_?[a-zA-Z0-9]+)*$"/&gt;
+ *    &lt;property name="allowClassName" value="false"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>Code Example:</p>
+ * <pre>
+ * class MyClass {
+ *   public MyClass() {} // OK
+ *   public void MyClass() {} // violation,  method Name 'MyClass' must not
+ *                            // equal the enclosing class name
+ * }
+ * </pre>
+ * <p>
+ * An example of how to suppress the check to public and protected member:
+ * </p>
+ * <pre>
+ * &lt;module name="MethodName"&gt;
+ *    &lt;property name="format" value="^[a-z](_?[a-zA-Z0-9]+)*$"/&gt;
+ *    &lt;property name="applyToPublic" value="false"&gt;
+ *    &lt;property name="applyToProtected" value="false"&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>Code Example:</p>
+ * <pre>
+ * class MyClass {
+ *   public void MyClass1() {} // OK
+ *   protected void MyClass2() {} // OK
+ *   private void MyClass3() {} // violation, name 'MyClass3' must match
+ *                              // pattern '^[a-z](_?[a-zA-Z0-9]+)*$'
```

Comment:
Since you won't show other examples of the `apply` in the configuration, I think we should just expand the code example to all types.
Please add a package-private method too.

Reply:
Done.